### ISSUE TITLE
Prettier dependency

### ISSUE THREAD

**Issue by bkniffler** [helios1138/graphql-typed-client#19]
Hi, awesome library, I think it boasts the best interface for graphql clients! 
Though, when building a lambda function, I noticed that 'prettier' is being used within the library and will thus be bundled by webpack. Is prettier really necessary in client runtime?

I'm asking, since prettier is very heavy and will make the build process considerably slower, at least in my case.

**Comment by bkniffler** [comment]
Also, lots of warnings pop open during build: 
```
3:13:25 AM: WARNING in /opt/build/repo/node_modules/prettier/third-party.js 4368:11-28
3:13:25 AM: Critical dependency: the request of a dependency is an expression
3:13:25 AM:  @ /opt/build/repo/node_modules/prettier/index.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/helpers/prettify.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/encoding/lib/iconv-loader.js 9:12-34
3:13:25 AM: Critical dependency: the request of a dependency is an expression
3:13:25 AM:  @ /opt/build/repo/node_modules/encoding/lib/encoding.js
3:13:25 AM:  @ /opt/build/repo/node_modules/isomorphic-fetch/node_modules/node-fetch/lib/body.js
3:13:25 AM:  @ /opt/build/repo/node_modules/isomorphic-fetch/node_modules/node-fetch/index.js
3:13:25 AM:  @ /opt/build/repo/node_modules/isomorphic-fetch/fetch-npm-node.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/prettier/index.js 42576:28-47
3:13:25 AM: Critical dependency: the request of a dependency is an expression
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/helpers/prettify.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/prettier/index.js 36132:7-46
3:13:25 AM: Critical dependency: the request of a dependency is an expression
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/helpers/prettify.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/prettier/index.js 10593:17-66
3:13:25 AM: Critical dependency: the request of a dependency is an expression
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/helpers/prettify.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/lib/buffer-util.js
3:13:25 AM: Module not found: Error: Can't resolve 'bufferutil' in '/opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/lib'
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/lib/buffer-util.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/lib/sender.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/index.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/getSubscriptionCreator.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/ws/lib/buffer-util.js
3:13:25 AM: Module not found: Error: Can't resolve 'bufferutil' in '/opt/build/repo/node_modules/ws/lib'
3:13:25 AM:  @ /opt/build/repo/node_modules/ws/lib/buffer-util.js
3:13:25 AM:  @ /opt/build/repo/node_modules/ws/lib/receiver.js
3:13:25 AM:  @ /opt/build/repo/node_modules/ws/index.js
3:13:25 AM:  @ /opt/build/repo/node_modules/subscriptions-transport-ws/dist/server.js
3:13:25 AM:  @ /opt/build/repo/node_modules/subscriptions-transport-ws/dist/index.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/getSubscriptionCreator.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/lib/validation.js
3:13:25 AM: Module not found: Error: Can't resolve 'utf-8-validate' in '/opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/lib'
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/lib/validation.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/lib/sender.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/node_modules/ws/index.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/getSubscriptionCreator.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/ws/lib/validation.js
3:13:25 AM: Module not found: Error: Can't resolve 'utf-8-validate' in '/opt/build/repo/node_modules/ws/lib'
3:13:25 AM:  @ /opt/build/repo/node_modules/ws/lib/validation.js
3:13:25 AM:  @ /opt/build/repo/node_modules/ws/lib/receiver.js
3:13:25 AM:  @ /opt/build/repo/node_modules/ws/index.js
3:13:25 AM:  @ /opt/build/repo/node_modules/subscriptions-transport-ws/dist/server.js
3:13:25 AM:  @ /opt/build/repo/node_modules/subscriptions-transport-ws/dist/index.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/getSubscriptionCreator.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/prettier/parser-glimmer.js 1:127570-127588
3:13:25 AM: require.extensions is not supported by webpack. Use a loader instead.
3:13:25 AM:  @ /opt/build/repo/node_modules/prettier/index.js 35988:13-40
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/helpers/prettify.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/prettier/parser-glimmer.js 1:127549-127567
3:13:25 AM: require.extensions is not supported by webpack. Use a loader instead.
3:13:25 AM:  @ /opt/build/repo/node_modules/prettier/index.js 35988:13-40
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/helpers/prettify.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
3:13:25 AM: WARNING in /opt/build/repo/node_modules/prettier/parser-glimmer.js 1:127607-127625
3:13:25 AM: require.extensions is not supported by webpack. Use a loader instead.
3:13:25 AM:  @ /opt/build/repo/node_modules/prettier/index.js 35988:13-40
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/helpers/prettify.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/client/createClient.js
3:13:25 AM:  @ /opt/build/repo/node_modules/graphql-typed-client/dist/index.js
```

**Comment by remorses** [comment]
i can't use this library in nextjs because of the prettier dependency.

build phase fails with
```
Module not found: Can't resolve 'fs' in '/Users/morse/Documents/GitHub/EFI-archive/nextjs/node_modules/prettier'
```

**Comment by remorses** [comment]
We could make a separate dependency used in the generated code, without the prettier dependency. 

The only used code from the main package are `Client, ClientOptions, createClient, linkTypeMap, Observable`

**Comment by bkniffler** [comment]
Yeah, I had originally cloned this project and gotten rid of prettier, though I don't remember how exactly (its been a while now). I know it worked fine and it wasn't too much trouble though. Since the project here doesn't seemed to be too active, I never bothered to open a PR.

**Comment by helios1138** [comment]
Yes, sorry for the long absence. Prettier is definitely not needed for the client lib, but since it's bundled with the generator it causes problems for frontend usage (it was originally created with only the backend usage in mind). I will take a look at what can be done tomorrow.

**Comment by helios1138** [comment]
I'm pretty sure that this is the issue
https://github.com/helios1138/graphql-typed-client/blob/ade0d1487fffdce4453273430ce636cbd4cb94b1/src/client/createClient.ts#L16
I think this can be easily removed and then tree-shaking in webpack should take care of not including prettier in the bundle, even if it will still be in included in dependencies if I remember correctly how the webpack works

**Comment by helios1138** [comment]
@bkniffler should be fixed in `1.7.3`. if you have a way to check if that issue is fixed in your setup, I would appreciate :)
